src/components/PopupWithForm.js

```diff
@@ -6,6 +6,16 @@ export default class PopupWithForm extends Popup {
     this._popupForm = this._popupEl.querySelector(".modal__form");
     this._inputList = this._popupEl.querySelectorAll(".modal__input");
     this._handleFormSubmit = handleFormSubmit;
+    this._submitButton = this._popupEl.querySelector(".modal__button");
+    this._submitButtonText = this._submitButton.textContent;
+  }
+
+  viewLoading(isLoading, loadingText = "Saving...") {
+    if (isLoading) {
+      this._submitButton.textContent = loadingText;
+    } else {
+      this._submitButton.textContent = this._submitButtonText;
+    }
   }
 
```

Comment:
Please, delete `e.target.reset();` from line 35

You need to **reset** the form only  in block `then` not to do that immediately after `submit` because when there is an error the popup will still be opened and the user can try to send the data again to the server without having to type the data into the inputs once more